src/code/padmgr.c

```diff
@@ -1,58 +1,797 @@
+ * ~0.4 seconds at 60 VI/sec and ~0.48 seconds at 50 VI/sec
+ *
+ * @param enable Array of u8 of length MAXCONTROLLERS containing either true or false to enable or disable rumble
+ *               for that controller
+ */
+void PadMgr_RumbleSet(u8 enable[MAXCONTROLLERS]) {
+    s32 i;
+
+    for (i = 0; i < MAXCONTROLLERS; i++) {
+        sPadMgrInstance->rumbleEnable[i] = enable[i];
+    }
+    sPadMgrInstance->rumbleOnTimer = 24;
+}
+
+s32 PadMgr_ControllerHasRumblePak(s32 port) {
+    return sPadMgrInstance->pakType[port] == CONT_PAK_RUMBLE;
+}
+
+void PadMgr_AdjustInput(Input* input) {
+    s32 plus;
+    s32 minus;
+    s32 pressX;
+    s32 pressX2;
+    s32 pressY;
+    s32 pressY2;
+    s32 pad1;
+    s32 pad2;
+    s32 pad3;
+    f32 pad4;
+    f32 newX;
+    f32 newY;
+    f32 magnitude;
+    f32 angle;
+    s8 curX = input->cur.stick_x;
+    s8 curY = input->cur.stick_y;
+
+    if (CHECK_BTN_ANY(input->press.button, BTN_RESET) || input->press.stick_x == 0) {
+        goto dummy;
+    dummy:;
+
+        input->press.stick_x = 61;
+        input->press.errno = -61;
+        input->press.stick_y = 63;
+        input->rel.errno = -63;
+    }
+    pressX = input->press.stick_x;
+    pressX2 = (s8)input->press.errno;
+    pressY = input->press.stick_y;
+    pressY2 = (s8)input->rel.errno;
+
+    if (CHECK_BTN_ANY(input->cur.button, BTN_RESET)) {
+        minus = curX - 7;
+        plus = curX + 7;
+
+        if (minus > 0) {
+            if (pressX < minus - 3) {
+                pressX = minus - 3;
+                input->press.stick_x = minus - 3;
+            }
+        } else if (plus < 0) {
+            if (pressX2 > plus + 3) {
+                pressX2 = plus + 3;
+                input->press.errno = plus + 3;
+            }
+        }
+
+        minus = curY - 7;
+        plus = curY + 7;
+
+        if (minus > 0) {
+            if (pressY < minus - 3) {
+                pressY = minus - 3;
+                input->press.stick_y = minus - 3;
+            }
+        } else if (plus < 0) {
+            if (pressY2 > plus + 3) {
+                pressY2 = plus + 3;
+                input->rel.errno = plus + 3;
+            }
+        }
+    }
+
+    minus = curX - 7;
+    plus = curX + 7;
+
+    if (minus > 0) {
+        newX = (pressX < minus) ? 1.0f : (f32)minus / pressX;
+    } else if (plus < 0) {
+        newX = (pressX2 > plus) ? -1.0f : -(f32)plus / pressX2;
+    } else {
+        newX = 0.0f;
+    }
+
+    minus = curY - 7;
+    plus = curY + 7;
+
+    if (minus > 0) {
+        newY = (pressY < minus) ? 1.0f : (f32)minus / pressY;
+    } else if (plus < 0) {
+        newY = (pressY2 > plus) ? -1.0f : -(f32)plus / pressY2;
+    } else {
+        newY = 0.0f;
+    }
+
+    magnitude = sqrtf(SQ(newX) + SQ(newY));
+    if (magnitude > 1.0f) {
+        magnitude = 1.0f;
+    }
+    angle = Math_Atan2F_XY(curY, -curX);
+    newX = -sinf(angle) * magnitude;
+    newY = cosf(angle) * magnitude;
```

Comment:
```suggestion
    if(1) {
        s32 pad;

        magnitude = sqrtf(SQ(newX) + SQ(newY));
        if (magnitude > 1.0f) {
            magnitude = 1.0f;
        }
        angle = Math_Atan2F_XY(curY, -curX);
        newX = -sinf(angle) * magnitude;
        newY = cosf(angle) * magnitude;
    }
```
This lets you remove the gotos, and seems reasonable based on debug. Couldn't use magnitude and angle for some reason as the scoped variable.